File: src/main/java/com/jiro4989/tkfm/model/PositionModel.java
package com.jiro4989.tkfm.model;

import javafx.beans.property.DoubleProperty;
import javafx.beans.property.SimpleDoubleProperty;

public class PositionModel {
  private final DoubleProperty x = new SimpleDoubleProperty(0.0);
  private final DoubleProperty y = new SimpleDoubleProperty(0.0);

  public PositionModel() {
    this(0, 0);
  }

  public PositionModel(double x, double y) {
    setX(x);
    setY(y);
  }

  public DoubleProperty xProperty() {
    return x;
  }

  public DoubleProperty yProperty() {
    return y;
  }

  public double getX() {
    return x.get();
  }

  public double getY() {
    return y.get();
  }

  public void setX(double x) {
    this.x.set(x);
  }

  public void setY(double y) {
    this.y.set(y);
  }
}

File: src/main/kotlin/com/jiro4989/tkfm/model/PositionModel.kt
package com.jiro4989.tkfm.model

import javafx.beans.property.DoubleProperty
import javafx.beans.property.SimpleDoubleProperty as DP

data class PositionModel(val xProperty: DoubleProperty, val yProperty: DoubleProperty) {
  var x
    get() = xProperty.get()
    set(value) = xProperty.set(value)
  var y
    get() = yProperty.get()
    set(value) = yProperty.set(value)
  constructor() : this(DP(0.0), DP(0.0))
  constructor(x: Double, y: Double) : this(DP(x), DP(y))

  operator fun div(scale: Double) = Pair(x / scale, y / scale)
}
